src/utility/time/TimeService.cpp

```diff
@@ -91,6 +91,52 @@ unsigned long TimeService::getTime()
 #endif
 }
 
+void TimeService::setTimeZoneData(int offset, unsigned int dst_until)
+{
+  DEBUG_INFO("ArduinoIoTCloudTCP::%s offset[%d] dst_unitl[%ul]", __FUNCTION__, offset, dst_until);
+  _timezone_offset = offset;
+  _timezone_dst_until = dst_until;
+}
+
+unsigned long TimeService::getLocalTime()
+{
+  unsigned long utc = getTime();
+  return utc + _timezone_offset;
+}
+
+unsigned long TimeService::getTimeFromString(const String& timeString)
+{
+  char s_month[5];
+  int month, day, year, hour, minutes, seconds;
+  struct tm t =
+  {
+    0 /* tm_sec   */,
+    0 /* tm_min   */,
+    0 /* tm_hour  */,
+    0 /* tm_mday  */,
+    0 /* tm_mon   */,
+    0 /* tm_year  */,
+    0 /* tm_wday  */,
+    0 /* tm_yday  */,
+    0 /* tm_isdst */
+  };
+  static const char month_names[] = "JanFebMarAprMayJunJulAugSepOctNovDec";
+
+  sscanf(timeString.c_str(), "%d %s %d %d:%d:%d", &year, s_month, &day, &hour, &minutes, &seconds);
```

Comment:
this function is public but there are no checks on the input
for example, if a user executes the following statement, the program will crash
`  Schedule::getTimeFromString("2021 November 01 17:00:00");  `

is this something we care about ?
also, I think that the time format should be documented somewhere

Reply:
yes, definitely. I was really unsure about this function, andas you can see here https://github.com/arduino-libraries/ArduinoIoTCloud/pull/284/commits/d317f5631b91f1479089ef519bc6c55b15b64c71 i had tried to use also std::get_time function to do this. Do you think it would be better?

I didn't wont to add much complex checks mostly because i think most people would use the widget to configure the schedule, but for sure i will add some lined to document how to use the function in the example and in the TimeService class

Reply:
or maybe jut provide a wrapper around std::get_time provigind the time string and also the information on how is formatted..

Reply:
>as you can see here d317f56 i had tried to use also std::get_time function to do this. Do you think it would be better?

yes, using a std function with a standard time format seems better to me 

IMO all other alternatives are also good as long as they are documented (also the one you already implemented)

Reply:
fixed with 3cf18b29bd81944820adb793468b2f1b8a8d55e5